designs/2019-update-default-ignore-patterns/README.md

```diff
@@ -0,0 +1,59 @@
+- Start Date: 2019-11-28
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: Toru Nagashima ([@mysticatea](https://github.com/mysticatea))
+
+# Update Default Ignore Patterns
+
+## Summary
+
+This proposal changes the default ignore patterns to include `node_modules` in subdirectories and to exclude `.eslintrc.js`.
+
+## Motivation
+
+- In monorepo style development, there are `node_modules`s in subdirectories. But the current default ignore patterns don't include it, so we have to configure ESLint with `node_modules` to ignore it.
+- Because ESLint ignores dotfiles, it doesn't lint `.eslintrc.js` by default. We have to configure ESLint with `!.eslintrc.js` to lint it.
+
+## Detailed Design
+
+This proposal changes the default ignore patterns:
+
+- (as-is) `.*`
+- (new) `!.eslintrc.js`
+- (change) `/node_modules/*` → `/**/node_modules/*`
+- (as-is) `/bower_components/*`
```

Comment:
what do you think also change `/bower_components/` like `node_modules`, for consistency?

Reply:
Good point.

Because I'm not sure if the `bower_components` directory in subdirectories is a common use-case or not, so I have left it as-is. I think that we have three directions for this.

1. We keep it as-is.
1. We change it to `/**/bower_components/*`.
1. We remove `/bower_components/*` in favor of the `ignorePatterns` property of configs.

In fact, the 3. is consistent with our stance -- we don't add exceptions anymore because we don't want to have the allowlist/denylist of all tools in the world.

Reply:
bower’s been deprecated/dead/discouraged for many years; you could probably remove it in the next major and nobody would notice. I don’t think you want to make any changes now; I’d vote 3, 1, 2, in that order.